### ISSUE TITLE
Cannot make projection ?   

### ISSUE THREAD

**Issue by Waxoussou** [manyuanrong/deno_mongo#34]
First, i'd like to thank you for the amazing job you've been doing. 

I was playing with Deno and Mongo and I just realized that it was impossible for me to make projection when returning data from a find query. 
`
    const productList = await products.find({  }, {name: 1, price:1 , _id:0})
`
the result of my productList variable contains all products with their price and their id too ...
Am I missing something or is it something that is just not working for the moment ?

**Comment by manyuanrong** [comment]
You can currently use `aggregate`

**Comment by Waxoussou** [comment]
> You can currently use `aggregate`

Thanks @manyuanrong 
I have tried to use aggregate , but still not working. (or something I do could be wrong ...)
Here How I did my aggregation : 
```
        const productList = await products.aggregation(
            { $group: { _id:  "$name" } }
        )

```
I tested first that it would work correctly on mongo (cli) and it does work as expected but on deno, I get that response :

```
TypeError: products.aggregation is not a function
    at getProducts (file:///Users/maxbook/Code/denoDiscovery/controllers/products.js:15:44)
    at dispatch (https://deno.land/x/oak/middleware.ts:35:32)
    at https://deno.land/x/oak/router.ts:416:18
    at dispatch (https://deno.land/x/oak/middleware.ts:35:32)
    at composedMiddleware (https://deno.land/x/oak/middleware.ts:41:12)
    at dispatch (https://deno.land/x/oak/router.ts:420:28)
    at dispatch (https://deno.land/x/oak/middleware.ts:35:32)
    at default (file:///Users/maxbook/Code/denoDiscovery/middlewares/errorHandler.js:4:15)
    at dispatch (https://deno.land/x/oak/middleware.ts:35:32)
    at Object.composedMiddleware [as middleware] (https://deno.land/x/oak/middleware.ts:41:12)
```

**Comment by radek3911** [comment]
> 
> 
> > You can currently use `aggregate`
> 
> Thanks @manyuanrong
> I have tried to use aggregate , but still not working. (or something I do could be wrong ...)
> Here How I did my aggregation :
> 
> ```
>         const productList = await products.aggregation(
>             { $group: { _id:  "$name" } }
>         )
> ```
> 
> I tested first that it would work correctly on mongo (cli) and it does work as expected but on deno, I get that response :
> 
> ```
> TypeError: products.aggregation is not a function
>     at getProducts (file:///Users/maxbook/Code/denoDiscovery/controllers/products.js:15:44)
>     at dispatch (https://deno.land/x/oak/middleware.ts:35:32)
>     at https://deno.land/x/oak/router.ts:416:18
>     at dispatch (https://deno.land/x/oak/middleware.ts:35:32)
>     at composedMiddleware (https://deno.land/x/oak/middleware.ts:41:12)
>     at dispatch (https://deno.land/x/oak/router.ts:420:28)
>     at dispatch (https://deno.land/x/oak/middleware.ts:35:32)
>     at default (file:///Users/maxbook/Code/denoDiscovery/middlewares/errorHandler.js:4:15)
>     at dispatch (https://deno.land/x/oak/middleware.ts:35:32)
>     at Object.composedMiddleware [as middleware] (https://deno.land/x/oak/middleware.ts:41:12)
> ```

Hi,

The function is aggregate() not aggregation(). I have made a PR #37 that fixes the typo in README.md.

**Comment by Waxoussou** [comment]
> Hi,
> 
> The function is aggregate() not aggregation(). I have made a PR #37 that fixes the typo in README.md.

Hi @radek3911 , thanks for that information. 
I did try with aggregate instead of aggregation (and have been triyng again since i red your message but unfortunately, that does not work either.
Here is what I get from the console : 
```
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: map, expected a sequence", line: 1, column: 63)', src/command/aggregation.rs:19:37
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[denon] app crashed - waiting for file changes before starting ...
```

**Comment by manyuanrong** [comment]
Can you provide some data samples and sample code?

**Comment by Waxoussou** [comment]
here is some data sample : 
```
{
    "_id": {
        "$oid": "5eca9b8d41b84920755766e4"
    },
    "name": "banana",
    "price": 12
},
{
    "_id": {
        "$oid": "5eca9b8d41b84920755766e5"
    },
    "name": "fraise",
    "price": 1.41
},
{
    "_id": {
        "$oid": "5ecab4ed0090019a00b58991"
    },
    "name": "bernard",
    "price": 2.23
}
```
nothing too.... complicated.
And for the my code , a simple route : 
routes.js :
`router.get('/api/v1/products', getProducts)`

products.js :
```
// @desc Get all products  
// @route GET/api/v1/products 
const getProducts = async ({ response }) => {
        const productList = await products.find({})
        response.body = {
            success: true,
            data: productList,
        }

}

```

**Comment by Waxoussou** [comment]
with aggregate : 
```
const getProducts = async ({ response }) => {
        const productList = await products.aggregate(
            { $group: { _id: "$name" } }
        )
        response.body = {
            success: true,
            data: productList,
        }}
```

**Comment by radek3911** [comment]
```js
 const getProducts = async ({ response }) => {
         const productList = await products.aggregate(
             { $match: { <query> } },
             { $group: { _id: "$name" } }
         )
         response.body = {
             success: true,
             data: productList,
         }}
 ```

You are missing the $match parameter.
https://docs.mongodb.com/manual/reference/operator/aggregation/match/#pipe._S_match

**Comment by Waxoussou** [comment]
indeed, that does work as expected now with the $match parameter. 
It does work without it in Mongo cli for example, but not with the deno driver I believe...
Thanks @radek3911 for that solution

**Comment by radek3911** [comment]
Hi, I have looked into it and as far as I can tell, the match parameter is enforced by the Rust mongo library. If you would like to make the match parameter optional you need to make new issue at the Rust project.

Have a nice day and please close the issue if you don't have any more questions.

**Comment by Waxoussou** [comment]
thanks @radek3911 . I 've got all the answers so far. 
Issue is closed
